src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_StringArrayParam.generated.txt

```diff
@@ -213,9 +253,90 @@ namespace Microsoft.AspNetCore.Http.Generated
             }
         }
 
+        private static async ValueTask<(bool, T?)> TryResolveBodyAsync<T>(HttpContext httpContext, LogOrThrowExceptionHelper logOrThrowExceptionHelper, bool allowEmpty, string parameterTypeName, string parameterName, bool isInferred = false)
+        {
+            var feature = httpContext.Features.Get<Microsoft.AspNetCore.Http.Features.IHttpRequestBodyDetectionFeature>();
+
+            if (feature?.CanHaveBody == true)
+            {
+                if (!httpContext.Request.HasJsonContentType())
+                {
+                    logOrThrowExceptionHelper.UnexpectedJsonContentType(httpContext.Request.ContentType);
+                    httpContext.Response.StatusCode = StatusCodes.Status415UnsupportedMediaType;
+                    return (false, default);
+                }
+                try
+                {
+                    var bodyValue = await httpContext.Request.ReadFromJsonAsync<T>();
```

Comment:
@captainsafia Why aren't we passing the JSON type info here?

Reply:
> Context for why: https://github.com/dotnet/aspnetcore/pull/45359

Are you identifying the perf ramifications of querying the DI container for the JsonSerializerOptions to resolve the type info on every request? If so, that shouldn't be a problem here since we can cache the type info for the parameter type in the same way that we do for the response type.